processor-api/src/main/java/com/synopsys/integration/alert/processor/api/filter/JobNotificationMapper.java

```diff
@@ -73,7 +73,15 @@ public DefaultJobNotificationExtractor(JobAccessor jobAccessor) {
                 set.add(filterableNotificationWrapper);
             }
         }
-        return groupedFilterableNotifications;
+
+        List<FilteredJobNotificationWrapper> filterableJobNotifications = new LinkedList<>();
+        for (Map.Entry<FilteredDistributionJobResponseModel, List<FilterableNotificationWrapper>> groupedEntry : groupedFilterableNotifications.entrySet()) {
+            FilteredDistributionJobResponseModel filteredJob = groupedEntry.getKey();
+            FilteredJobNotificationWrapper wrappedJobNotifications = new FilteredJobNotificationWrapper(filteredJob.getId(), filteredJob.getProcessingType(), filteredJob.getChannelName(), groupedEntry.getValue());
+            filterableJobNotifications.add(wrappedJobNotifications);
+        }
```

Comment:
are we sure we want this additional processing into a list? It looks like an extra iteration for not too much of a benefit

Reply:
The benefit is a clearer API. We could optimize down to one iteration, but the concern this is addressing is to no longer return a `Map`.